src/main/preload/externalAPI.ts

```diff
@@ -0,0 +1,459 @@
+    const channel = channels.get(channelId) ?? {id: channelId};
+    channels.delete(channelId);
+    window.postMessage(
+        {
+            type: NOTIFICATION_CLICKED,
+            message: {
+                channel,
+                teamId,
+                url,
+            },
+        },
+        window.location.origin,
+    );
+});
+
+ipcRenderer.on(BROWSER_HISTORY_PUSH, (event, pathName) => {
+    window.postMessage(
+        {
+            type: 'browser-history-push-return',
+            message: {
+                pathName,
+            },
+        },
+        window.location.origin,
+    );
+});
+
+const sendHistoryButtonReturn = (status: {canGoBack: boolean; canGoForward: boolean}) => {
+    window.postMessage(
+        {
+            type: 'history-button-return',
+            message: {
+                enableBack: status.canGoBack,
+                enableForward: status.canGoForward,
+            },
+        },
+        window.location.origin,
+    );
+};
+
+ipcRenderer.on(BROWSER_HISTORY_STATUS_UPDATED, (event, canGoBack, canGoForward) => sendHistoryButtonReturn({canGoBack, canGoForward}));
+
+const sendDesktopSourcesResult = (sources: Array<{
+    id: string;
+    name: string;
+    thumbnailURL: string;
+}>) => {
+    window.postMessage(
+        {
+            type: DESKTOP_SOURCES_RESULT,
+            message: sources,
+        },
+        window.location.origin,
+    );
+};
+
+const sendCallsJoinedCall = (callID: string) => {
+    window.postMessage(
+        {
+            type: CALLS_JOINED_CALL,
+            message: {callID},
+        },
+        window.location.origin,
+    );
+};
+
+ipcRenderer.on(CALLS_JOIN_REQUEST, (_, callID) => {
+    window.postMessage(
+        {
+            type: CALLS_JOIN_REQUEST,
+            message: {callID},
+        },
+        window.location.origin,
+    );
+});
+
+ipcRenderer.on(DESKTOP_SOURCES_MODAL_REQUEST, () => {
+    window.postMessage(
+        {
+            type: DESKTOP_SOURCES_MODAL_REQUEST,
+        },
+        window.location.origin,
+    );
+});
+
+ipcRenderer.on(CALLS_WIDGET_SHARE_SCREEN, (_, sourceID, withAudio) => {
+    window.postMessage(
+        {
+            type: CALLS_WIDGET_SHARE_SCREEN,
+            message: {sourceID, withAudio},
+        },
+        window.location.origin,
+    );
+});
+
+ipcRenderer.on(CALLS_ERROR, (_, err, callID, errMsg) => {
+    window.postMessage(
+        {
+            type: CALLS_ERROR,
+            message: {err, callID, errMsg},
+        },
+        window.location.origin,
+    );
+});
+
+// push user activity updates to the webapp
+ipcRenderer.on(USER_ACTIVITY_UPDATE, (event, userIsActive, isSystemEvent) => {
+    if (window.location.origin !== 'null') {
+        window.postMessage({type: USER_ACTIVITY_UPDATE, message: {userIsActive, manual: isSystemEvent}}, window.location.origin);
+    }
+});
+
+/**
+ * Legacy functionality that needs to be disabled with the new API
```

Comment:
There's a bunch of functionality that requires listeners or some `setIntervals` to be run to work correctly with legacy Web App versions, but this code might have adverse effects for newer Web App versions.

So here we have a switch to completely turn those off, which is called when we do the initial handshake between the Desktop/Web Apps.